delocate/libsana.py

```diff
@@ -104,6 +107,55 @@ def resolve_rpath(lib_path, rpaths):
     return lib_path
 
 
+def search_environment_for_lib(lib_path):
+    """ Search common environment variables for `lib_path`
+
+    Apple's documentation about how to do this does not quite match what we see
+    in the wild, so we'll use a single approach here:
+
+        1. Search for the basename of the library on DYLD_LIBRARY_PATH
+        2. Search for ``realpath(lib_path)``
+        3. Search for the basename of the library on DYLD_FALLBACK_LIBRARY_PATH
+
+    Testing has suggested that LD_LIBRARY_PATH is not actually used in this
```

Comment:
Is that also true for binaries compiled with gcc?  I guess so.

Reply:
Hm, I'm not sure -- I'll try and check. I can't remember off the top of my head if I was using clang or gcc.

Reply:
Did you get round to checking this?   The ideal would be some test snippets to put in repository, but outside the main code, to confirm the behavior.

Reply:
Agreed -- I'll put a script in `tests/data` that walks thorugh this, like you have with `test-libs`.

With reference to your original comment, I'm not sure I completely understand it -- why would you expect the compiler to change the way that `@rpath` entries are resolved? Those entries should be resolved at run-time, not compile-time.

Reply:
Ok, so I've done some testing and put a bit of a demonstration in the `testing_osx_rpath_env_variables.sh` script above. I didn't quite have this right: if the library that's being searched for has a directory in its path (which seems to me like it's usually going to be the case, unless it's being referenced by something that's in the same folder as you're delocating things in), then the search order is like I laid out above: check DYLD, then try the full path to the lib, and then try DYLD_FALLBACK. If it *doesn't* have a directory in its path, then you check `LD_LIBRARY_PATH` first, then DYLD, then you search for the lib in the directory you're currently in, and then you search DYLD_FALLBACK. I could add that here as an option if you like.

Reply:
Went for just stripping out `LD_LIBRARY_PATH` to simplify things.